res/res_usbradio.c

```diff
@@ -820,19 +858,108 @@ void ast_radio_print_rx_audio_stats(int fd, struct rxaudiostatistics *o)
 	}
 }
 
+/*!
+ * \brief Remove and free up all user devices.
+ */
+static void cleanup_user_devices(void)
+{
+	struct user_usb_device *device;
+	/* Remove all existing devices */
+	AST_RWLIST_WRLOCK(&user_devices);
+	AST_LIST_TRAVERSE_SAFE_BEGIN(&user_devices, device, entry) {
+		AST_LIST_REMOVE_CURRENT(entry);
+		ast_free(device);
+	}
+	AST_LIST_TRAVERSE_SAFE_END;
+	AST_RWLIST_UNLOCK(&user_devices);
+}
+
+/* Load our configuration */
+static int load_config(int reload)
+{
+	struct ast_config *cfg;
+	struct ast_flags config_flags = { reload ? CONFIG_FLAG_FILEUNCHANGED : 0 };
+	const char *varval;
+
+	if (!(cfg = ast_config_load(CONFIG_FILE, config_flags))) {
+		ast_log(LOG_WARNING, "Config file %s not found\n", CONFIG_FILE);
+		return 0;
+	} else if (cfg == CONFIG_STATUS_FILEUNCHANGED) {
+		ast_debug(1, "Config file %s unchanged, skipping\n", CONFIG_FILE);
+		return 0;
+	} else if (cfg == CONFIG_STATUS_FILEINVALID) {
+		ast_log(LOG_ERROR, "Config file %s is in an invalid format. Aborting.\n", CONFIG_FILE);
+		return -1;
+	}
+	
+	if (reload) {
+		cleanup_user_devices();
+	}
+	
+	/* general section 
+	 * usb_devices format vvvv:pppp,vvvv:pppp where vvvv is usb vendor id and pppp is usb product id
+	 */
+	if ((varval = ast_variable_retrieve(cfg, "general", "usb_devices")) && !ast_strlen_zero(varval)) {
+		struct user_usb_device *device;
+		char *item;
+		char *value;
+		int idVendor;
+		int idProduct;
+		
+		value = ast_strdupa(varval);
+		
+		/* process the delimited list */
+		while ((item = ast_strsep(&value, ',', AST_STRSEP_STRIP))) {

```

Comment:
Curious, why ast_strsep? I seem to recall there being a downside to using this instead of strsep, but don't remember what it is right now.

Reply:
I changed this to strsep.